EmotiBitOscilloscope/src/ofApp.cpp

```diff
@@ -77,148 +76,219 @@ void ofApp::update() {
 
 void ofApp::checkLatestSwVersion()
 {
-	bool newVersionAvailable = false;
-	// system call to curl
-	std::string latestReleaseUrl = "https://github.com/EmotiBit/ofxEmotiBit/releases/latest";
-	std::string latestReleaseApiRequest = "https://api.github.com/repos/EmotiBit/ofxEmotiBit/releases/latest";
-	std::string command = "curl " + latestReleaseApiRequest;
-	std::string response = "";
+
+	bool isNetworkAvailable = false;
+	// -n specifies number of tries. we are going to try to detect internet connection by sending 1 packet
+	// -w specifies timeout period in mS. We are going to wait for a maximum of 3 seconds for a ping response.
+	// the IP 8.8.8.8 is google's public DNS. We are using ping with a specific IP as that removes undefined wait periods associated with DNS resolution. More details: https://developers.google.com/speed/public-dns
+	std::string command = "ping -n 1 -w 3000 8.8.8.8";
+	std::string pingResponse = "";
 	char buffer[200];
-	bool exceptionOccured = false;
+	ofLogNotice() << "Trying to detect internet access";
 	try
 	{
 #if defined (TARGET_OSX) || defined (TARGET_LINUX)
 		FILE* pipe = popen(command.c_str(), "r"); // returns NULL on fail. refer: https://man7.org/linux/man-pages/man3/popen.3.html#RETURN_VALUE
 #else
 		FILE* pipe = _popen(command.c_str(), "r"); // returns NULL on fail. refer: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/popen-wpopen?view=msvc-170#return-value
-#endif
+#endif		
 		if (pipe != NULL)
 		{
-			try
+			while (fgets(buffer, sizeof buffer, pipe) != NULL)
 			{
-				while (fgets(buffer, sizeof buffer, pipe) != NULL)
+				pingResponse += buffer;
+			}
+
+#if defined (TARGET_OSX) || defined (TARGET_LINUX)
+			pclose(pipe);
+#else
+			_pclose(pipe);
+#endif
+		}
+		else
+		{
+			ofLogNotice() << "Failed to open pipe. ping failed";
+		}
+	}
+	catch (...)
+	{
+		ofLogError() << "Ping failed";
+	}
+	ofLogNotice() << pingResponse.c_str();
+	
+	// A successful ping response contains a sumary of round trip times. We are using that do determine if ping was successful
+	/* A successful ping response
+	------------------------------
+	Pinging 8.8.8.8 with 32 bytes of data:
+	Reply from 8.8.8.8: bytes = 32 time = 18ms TTL = 119
+
+		Ping statistics for 8.8.8.8 :
+		Packets : Sent = 1, Received = 1, Lost = 0 (0 % loss),
+		Approximate round trip times in milli - seconds :
+		Minimum = 18ms, Maximum = 18ms, Average = 18ms
+	*/
+	if (pingResponse.find("Approximate round trip times") != std::string::npos)
+	{
+		// substring found!
+		ofLogNotice() << "internet access detected";
+		isNetworkAvailable = true;
+	}
+	else
+	{
+		// failed
+		ofLogNotice() << "Failed to ping IP";
+	}
+
+	if (isNetworkAvailable)
```

Comment:
For reviewer:
All changes below are just indentation changes.
the source code remains the same.